tests/runtime/abi.ts

```diff
@@ -28,4 +75,25 @@ export class Abi {
     const bytes = this.heap.subarray(start, start + len * 2)
     return this.text.decoder.decode(bytes)
   }
+
+  public writeArrayBuffer(value: ArrayBuffer): Pointer {
+    const ptr = this.allocate(8 + value.byteLength)
+
+    this.view.setUint32(ptr, value.byteLength, LE)
+    this.heap.set(new Uint8Array(value), ptr + 8)
+
+    return ptr
+  }
+
+  public writeBigInt(value: bigint): Pointer {
+    const bytes = toBytesLE(value)
+    const bufferPtr = this.writeArrayBuffer(bytes)
+
+    const ptr = this.allocate(12)
+    this.view.setUint32(ptr, bufferPtr, LE)
+    this.view.setUint32(ptr + 4, 0, LE)
```

Comment:
Should this use `byteOffset` for consistency?

Reply:
Great catch! This code assumes that the `Uint8Array` always starts at the beginning of the `ArrayBuffer`. While this is true for `BigInt`, it isn't true in general (for example user specified `Uin8Array` event data for example).